GoTrue/src/commonMain/kotlin/io/github/jan/supabase/gotrue/AccessToken.kt

```diff
@@ -0,0 +1,36 @@
+package io.github.jan.supabase.gotrue
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.annotations.SupabaseInternal
+import io.github.jan.supabase.plugins.MainConfig
+import io.github.jan.supabase.plugins.MainPlugin
+
+/**
+ * Returns the access token used for requests. The token is resolved in the following order:
+ * 1. [jwtToken] if not null
+ * 2. [SupabaseClient.customAccessToken] if not null
+ * 3. [Auth.currentAccessTokenOrNull] if the Auth plugin is installed
+ * 4. [SupabaseClient.supabaseKey] if [keyAsFallback] is true
+ */
+@SupabaseInternal
+suspend fun SupabaseClient.accessToken(
+    jwtToken: String? = null,
+    keyAsFallback: Boolean = true
```

Comment:
Do we need these two parameters? They are not present on the js lib, right?

Reply:
This method is just an internal utility function, hence the `@SupabaseInternal` annotation. The actual public api is exactly the same as JS (located in the SupabaseClientBuilder):
```kotlin
val supabase = createSupabaseClient(supabaseUrl, supabaseKey) {
    accessToken = {
          //fetch the third party token
         "my-token"
    }
}
```

Reply:
And this method can be accessed via `supabase.customAccessToken()`, but we can rename this if this fits better.

Reply:
> This method is just an internal utility function

Oh yeah, what was I looking at 😂

> we can rename this if this fits better.

I think it's fine to just keep it in!